deeppavlov/models/embedders/elmo_embedder.py

```diff
@@ -129,40 +153,63 @@ def __call__(self, batch: List[List[str]],
 
         batch = filled_batch
 
+        if self.max_token:
+            batch = [batch_line[:self.max_token] for batch_line in batch]
         tokens_length = [len(batch_line) for batch_line in batch]
         tokens_length_max = max(tokens_length)
         batch = [batch_line + ['']*(tokens_length_max - len(batch_line)) for batch_line in batch]
 
-        elmo_outputs = self.sess.run(
-                                    self.elmo_outputs,
-                                    feed_dict=
-                                    {
-                                        self.tokens_ph: batch,
-                                        self.tokens_length_ph: tokens_length,
-                                    }
-                                    )
+        return batch, tokens_length
 
-        if self.mean:
-            batch = elmo_outputs['default']
+    @overrides
+    def __call__(self, batch: List[List[str]],
+                 *args, **kwargs) -> Union[List[np.ndarray], np.ndarray]:
+        """
+        Embed sentences from a batch.
 
-            dim0, dim1 = batch.shape
+        Args:
+            batch: A list of tokenized text samples.
 
+        Returns:
+            A batch of ELMo embeddings.
+        """
+        batch, tokens_length = self._fill_batch(batch)
+
+        elmo_outputs = self.sess.run(self.elmo_outputs,
+                                     feed_dict={self.tokens_ph: batch,
+                                                self.tokens_length_ph: tokens_length})
+
+        if 'default' in self.elmo_output_names:
+            elmo_output_values = elmo_outputs['default']
+            dim0, dim1 = elmo_output_values.shape
             if self.dim != dim1:
-                batch = np.resize(batch, (dim0,self.dim))
+                shape = (dim0, self.dim if isinstance(self.dim, int) else self.dim[0])
+                elmo_output_values = np.resize(elmo_output_values, shape)
         else:
-            batch = elmo_outputs['elmo']
+            elmo_output_values = list(map(lambda elmo_output_name: elmo_outputs[elmo_output_name], self.elmo_output_names))
```

Comment:
A list comprehension would look better here

Reply:
it is updated in 9efeba2 commit.